Library/Homebrew/download_strategy.rb

```diff
@@ -273,6 +273,7 @@ def ext
 
 class CurlDownloadStrategy < AbstractFileDownloadStrategy
   attr_reader :mirrors, :tarball_path, :temporary_path
+  attr_accessor :tap
```

Comment:
I fail to see why you need to know the tap. And it's super weird to have non-bottle url contains "bottle". Even more, as I said, this value is never set. So it will be "bottle" for any tap.

Why cannot you just name your Artifactoy repo to e.g. `Homebrew`. And use it by:
```
export HOMEBREW_ARTIFACT_BOTTLE=https://example.com/artifactory/homebrew
```